src/Plugin/migrate/source/YAML.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace Drupal\migrate_default_content\Plugin\migrate\source;
+
+use Drupal\migrate\MigrateException;
+use Drupal\migrate\Plugin\migrate\source\SourcePluginBase;
+use Drupal\migrate\Plugin\MigrationInterface;
+
+/**
+ * Source for YAML.
+ *
+ * If the YAML file contains non-ASCII characters, make sure it includes a
+ * UTF BOM (Byte Order Marker) so they are interpreted correctly.
+ *
+ * @MigrateSource(
+ *   id = "yaml"
```

Comment:
Ok, let's keep this commit in scope. The source plugin can live in this module and we can create a followup to separate it and create a dependency.

BTW. This is the process to create a full project
https://www.drupal.org/node/1011698
I went through it and if you want I can crete the project and then add you as a maintainer.